### ISSUE TITLE
Allow task providers to validate tasks and provide diagnostics

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#32830]
TL;DR - Feature request: have a way for task providers to report diagnostics on tasks in `tasks.json`.

Currently a `TaskProvider` only has two functions: `provideTasks` and `resolveTasks`.

Let's set aside `provideTasks` and consider the case of User-configured tasks in `tasks.json`.

For each User-configured task, `resolveTasks` is called to resolve it and create a `ProcessExecution | ShellExecution | CustomExecution` capable of running it. `resolveTasks` is not called exclusively when running a task. It can be called multiple times before to check if User-configured tasks can be handled by the extension.

So `resolveTasks` is a place where the task provider can detect errors that prevent the execution of the task and that cannot be detected solely by the JSON schema. E.g. a task property could reference a non-existing file - this cannot be detected by the JSON schema and can be checked in `resolveTasks`.

However the problem is that `resolveTasks` has no means to provide diagnostics. It can either return `undefined` or throw an error. In both cases a generic message is shown: `There are task errors. See the output for details.`, and in the output the error is generic and not meaningful to Users:

```
Error: The <type> task detection didn't contribute a task for the following configuration:
...
The task will be ignored.
```

There is the possibility of using `vscode.window.showErrorMessage()` to display an error, however since `resolveTasks` can be called multiple times, and sometimes not in the purpose of running the task, multiple errors might be displayed at a time where the User didn't ask to run the task which isn't ideal.

So all in all task providers currently don't have a good way of reporting invalid tasks.

The feature that I'd like is for the API to provide the means for a task provider to report diagnostics on tasks, but I haven't thought of a concrete proposal of what it would look like. TBD if this feature request gathers momentum.
